edb/edgeql/compiler/inference/cardinality.py

```diff
@@ -55,6 +55,14 @@
 AT_LEAST_ONE = qltypes.Cardinality.AT_LEAST_ONE
 
 
+class CardCtx(NamedTuple):
+    env: context.Environment
+    inferred_cardinality: Dict[
+        Tuple[irast.Base, irast.ScopeTreeNode],
+        qltypes.Cardinality]
+    singletons: Collection[irast.PathId]
```

Comment:
Am I missing something or is `singletons` not used anymore?

Reply:
Yeah, it doesn't seem to be used. One of the goals of this is so that changes like removing singletons could be less noisy. I could drop it in this PR or the next.